datarobot_batch_scoring/writer.py

```diff
@@ -454,18 +454,10 @@ def format_result_data(self, result, batch):
                 written_fields = out_fields[1:]
                 comb = [row[1:] for row in pred]
 
-        return (written_fields, comb)
-
-    def sigterm_handler(self, _signo, _stack_frame):
```

Comment:
I think removing signal handling is a mistake. Without it you cannot perform a cleanup if anything besides keyboardinterupt/SIGINT is received by the master. 
If you SIGKILL the master, the workers keep going like nothing happened because the master dies immediately without any cleanup. 
If you SIGKILL/SIGINT a workers, it just dies without  triggering a program shutdown.  The program only exits much later.
Every process should have a way to catch signals and trigger a fast, but safe, shutdown.

If they SIGKILL they are on their own :)